packages/kbn-esql-ast/src/visitor/README.md

```diff
@@ -0,0 +1,86 @@
+## High-level AST structure
+
+Broadly, there are two AST node types: (1) commands (say `FROM ...`, like
+*statements* in other languages), and (2) expressions (say `a + b`, or `fn()`).
+
+
+### Commands
+
+Commands in ES|QL are like *statements* in other languages. They are the top
+level nodes in the AST.
+
+The root node of the AST is considered to bye the "query" node. It contains a
+list of commands.
+
+```
+Quey = Command[]
+```
+
+Each command receives a list of positional arguments. For example:
+
+```
+COMMAND arg1, arg2, arg3
+```
+
+A command may also receive additional lists of *named* arguments, we refer to
+them as `option`s. For example:
+
+```
+COMMAND arg1, arg2, arg3 OPTION1 arg4, arg5 OPTION2 arg6, arg7
+```
+
+Essentially, one can of command arguments as a list of expressions, with the
+ability to add named arguments to the command. For example, the above command
+arguments can be represented as:
+
+```js
+{
+  '': [arg1, arg2, arg3],
+  'option1': [arg4, arg5],
+  'option2': [arg6, arg7]
+}
+```
+
+#### Command list
+
+Below is a list of know commands as of this writing:
+
+- `EXPLAIN`
+- `FROM <sources> [ METADATA <columns> ]`
+- `ROW <columns>`
+- `METRICS <sources> [ <columns> [ BY <columns> ]]`
+- `SHOW <identifier>`
+- `META <identifier>`
+- `EVAL <columns>`
+- `STATS <columns> [ BY <columns> ]`
+- `INLINESTATS <columns> [ BY <columns> ]`
+- `LIMIT <literal>`
+- `LOOKUP <source> ON <column>`
+- `KEEP <columns>`
+- `SORT <columns>`
+- `WHERE <expression>`
+- `DROP <columns>`
+- `RENAME <column> AS <column>`
+- `DISSECT <column> <string> [ APPEND_SEPARATOR = <string> ]`
+- `GROK <column> <string>`
+- `ENRICH <column> [ ON <column> ] [ WITH <columns> ]`
+- `MV_EXPAND <column>`
```

Comment:
This is a nice list, but I wonder if we just link to the docs since it may change in the future?

Reply:
Yes, I'll remove it.